### DISCUSSION TITLE
Feature request: implementation of eth_getRawTransactionByHash

### DISCUSSION THREAD

**Discussion by MagicalTux** [0xPolygon/polygon-edge#664] [Ideas]
### Description

Ethereum RPC API has `eth_getRawTransactionByHash` which can be useful to debug issues/etc. The method simply returns a transaction's raw bytes as stored on the blockchain and should be fairly simple to implement. 

### Your environment

* OS and version: Linux
* version of the Polygon Edge: v0.4.1

### Expected behaviour

```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": "0x02..."
}
```

### Actual behaviour

```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "error": {
    "code": -32601,
    "message": "the method eth_getRawTransactionByHash does not exist/is not available"
  }
}
````

### Relevant code

https://github.com/0xPolygon/polygon-edge/blob/develop/jsonrpc/eth_endpoint.go#L230

**Comment by zivkovicmilos** [comment]
Hey @MagicalTux,

Thank you for opening this discussion.

Why would you want this type of endpoint? For what would you use it for?

This is not a standard JSON-RPC Ethereum endpoint

**Reply by MagicalTux** [reply]
Such an endpoint can be useful to test and find issues when developing eth rpc clients to see what was exactly sent for a transaction as RLP and optionally when transactions are EIP 1159/etc (the same transaction can be represented different ways/etc).

While this endpoint is not standardized, it is useful and has a very low overhead (only need to return raw transaction but apart from that it works similar to `eth_getTransactionByHash`).

Of course the transaction can also be re-created client-side from `eth_getTransactionByHash` but that can become tricky as more eip-2718 transaction types are implemented in the future.

In some cases seeing how a transaction was encoded can be useful for various purposes too (learning, debugging, developing, etc).
